fs-storage/src/base_storage.rs

```diff
@@ -0,0 +1,34 @@
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::str::FromStr;
+
+use data_error::Result;
+
+pub trait BaseStorage<K, V>
+where
+    K: FromStr + Hash + Eq + Ord + Debug + Clone,
+    V: Debug + Clone,
+{
+    fn get(&self, id: &K) -> Option<V>;
```

Comment:
Since `BaseStorage` is an abstraction of top of a file/files that store key value mappings. There might be a way to derive [AsRef](https://doc.rust-lang.org/std/convert/trait.AsRef.html) for BaseStorage and to get most immutable BtreeMap related functions for free.

Reply:
Notice that we don't want to derive AsRefMut because we have some custom logic to update timestamp when doing mutable operations.